python/lsst/jointcal/utils.py

```diff
@@ -51,39 +51,47 @@ def __init__(self, match_radius=0.1*arcseconds, flux_limit=100.0, verbose=False)
         self.verbose = verbose
         self.log = lsst.log.Log.getLogger('JointcalStatistics')
 
-    def compute_rms(self, data_refs, visit_list, reference):
+    def compute_rms(self, data_refs, reference):
         """
         Match all data_refs to compute the RMS, for all detections above self.flux_limit.
 
         Parameters
         ----------
         data_refs : list of lsst.daf.persistence.butlerSubset.ButlerDataRef
             A list of data refs to do the calculations between.
-        visit_list : list of visit id (usually int)
-            list of visit identifiers to do the catalog merge on.
         reference : lsst reference catalog
             reference catalog to do absolute matching against.
 
         Return
         ------
-        astropy.Quantity
-            New relative RMS of the matched sources.
-        astropy.Quantity
-            New absolute RMS of matched sources.
+        namedtuple:
+            astropy.Quantity
+                Post-jointcal relative RMS of the matched sources.
+            astropy.Quantity
+                Post-jointcal absolute RMS of matched sources.
+            float
+                Post-jointcal photometric repeatability (PA1 from the SRD).
         """
 
         # DECAM doesn't have "filter" in its registry, so we have to get filter names from VisitInfo.
         self.filters = [ref.get('calexp').getInfo().getFilter().getName() for ref in data_refs]
-        visits_per_dataRef = [ref.dataId['visit'] for ref in data_refs]
+        self.visits_per_dataRef = [ref.dataId['visit'] for ref in data_refs]
 
         def compute(catalogs, calibs):
             """Compute the relative and absolute matches in distance and flux."""
-            visit_catalogs = self._make_visit_catalogs(catalogs, visits_per_dataRef, visit_list)
-            catalogs = list(visit_catalogs.values())
+            visit_catalogs = self._make_visit_catalogs(catalogs, self.visits_per_dataRef)
+            catalogs = [visit_catalogs[x] for x in self.visits_per_dataRef]
             # use the first catalog as the relative reference catalog
+            # NOTE: The "first" catalog depends on the original ordering of the data_refs.
+            # NOTE: Thus, because I'm doing a many-1 match in _make_match_dict,
+            # the number of matches (and thus the details of the match statistics)
+            # will change if the data_refs are ordered differently.
+            # All the more reason to use a proper n-way matcher here...
```

Comment:
I only have the epic for reworking it (which I've added a reference to): I don't see any tickets about actually writing an n-way matcher.